Library/Services/Implementation/NfieldSurveyInvitationTemplatesService.cs

```diff
@@ -39,6 +50,38 @@ public Task<IEnumerable<InvitationTemplateModel>> GetAsync(string surveyId)
                 .FlattenExceptions();
         }
 
+        public Task<InvitationTemplateModel> UpdateAsync(string surveyId, InvitationTemplateModel invitationTemplate)
+        {
+            CheckRequiredStringArgument(surveyId, nameof(surveyId));
+
+            var uri = SurveyInvitationTemplatesUrl(surveyId);
+
+            var updatedInvitationTemplate = new UpdateInvitationTemplate
+            {
+                Body = invitationTemplate.Body,
+                Subject = invitationTemplate.Subject,
+                Name = invitationTemplate.Name,
+                InvitationType = invitationTemplate.InvitationType
+            };
+
+            return Client.PatchAsJsonAsync(uri + "/" + invitationTemplate.Id, updatedInvitationTemplate)
+                .ContinueWith(responseMessageTask => responseMessageTask.Result.Content.ReadAsStringAsync().Result)
+                .ContinueWith(stringTask => JsonConvert.DeserializeObject<InvitationTemplateModel>(stringTask.Result))
+                .FlattenExceptions();
+        }
+
+        public Task<bool> RemoveAsync(string surveyId, string templateId)
```

Comment:
I replaced it with the invitationTemplate itself, because that's the thing you want to remove